packages/block-library/src/footnotes/index.php

```diff
@@ -26,13 +26,26 @@ function render_block_core_footnotes( $attributes, $content, $block ) {
 		return;
 	}
 
-	$footnotes = get_post_meta( $block->context['postId'], 'footnotes', true );
+	$raw_footnotes = get_post_meta( $block->context['postId'], 'footnotes', true );
 
-	if ( ! $footnotes ) {
+	if ( ! $raw_footnotes ) {
 		return;
 	}
 
-	$footnotes = json_decode( $footnotes, true );
+	$footnotes = json_decode( $raw_footnotes, true );
+
+	// The post meta contains malformed JSON. See
+	// https://core.trac.wordpress.org/ticket/59103.
+	if ( ! is_array( $footnotes ) ) {
+		// This will slash everything, including the keys value boundaries.
+		$footnotes = wp_slash( $raw_footnotes );
+		// So we need to unslash the keys, which are limited to "content" and
+		// "id".
+		$footnotes = str_replace( '{\\"content\\":\\"', '{"content":"', $footnotes );
+		$footnotes = str_replace( '\\",\\"id\\":\\"', '","id":"', $footnotes );
+		$footnotes = str_replace( '\\"}', '"}', $footnotes );
```

Comment:
🤔 json_decode below is still returning `null` for me. 

Test string is `[{"content":""","id":"237352bf-7f09-4346-85a8-7cc18ae8e70e"},{"content":"asdfasdf\" \"\" a''","id":"ce13a0ab-c2cc-4c9c-92ed-00a7b1b19228"}]`

Maybe it's an edge case, I'm not sure.

At this point, the value of `$footnotes` is `[{"content":"\"","id":"237352bf-7f09-4346-85a8-7cc18ae8e70e"},{"content":"asdfasdf\\\" \\\"\\\" a\'\'","id":"ce13a0ab-c2cc-4c9c-92ed-00a7b1b19228"}]`

It's the `\\\"` in the second footnote that seems to be the problem.

Reply:
> Maybe it's an edge case, I'm not sure.

I don't think it's an edge case, it seems to happen for me, too in any footnote that reaches this path containing a single quote character. I think it's because `\'` winds up causing a syntax error for the JSON decoder. Here's some test markup:

```
[{"content":"don't "break" with single quotes","id":"237352bf-7f09-4346-85a8-7cc18ae8e70e"}]
```

When processed by this code, this then becomes:

```
[{"content":"don\'t \"break\" with single quotes","id":"237352bf-7f09-4346-85a8-7cc18ae8e70e"}]
```

(Note the unexpected `\'` in "don\'t")

Interestingly, that kind of JSON will break in JS, too:

<img width="709" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/72facdd2-0ffd-4bfd-a22e-dd1e12d2e69b">

A fix appears to be to add the following before the `json_decode` line below:

```
$footnotes = str_replace( "\\'", "'", $footnotes );
```

That appears to get things working on the site frontend for me (kudos @tserong for the one-liner).

Reply:
@ramonjd @andrewserong Confirmed the problem. I replaced `wp_slash` with a `str_replace` instead of adding another `str_replace`, see 6343d0cf6eb13cc9eceba966ae8d7c20c1ffb8bb.